WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,236 @@
+import Foundation
+import CryptoKit
+
+/// An LRU disk cache that stores data in files on disk.
+///
+/// ``DiskCache`` uses an LRU cleanup policy where the least recently used items
+/// are removed first). The elements stored in the cache are automatically
+/// discarded if either *cost* or *count* limit is exceeded.
+///
+/// ``DiskCache`` is thread-safe; both reads and writes can be executed in parallel.
+///
+/// - important: It's not recommended to have more than one instance of ``DiskCache``
+/// managing the same path.
+final class DiskCache {
+    static let shared = DiskCache(
+        url: URL.getCachesURL().appendingPathComponent("WPCache", isDirectory: true)
+    )
+
+    private let rootURL: URL
+    private let sizeLimit: Int
+    private let sweepInterval: TimeInterval = 86400 // Around 1 day
+
+    private let queue = DispatchQueue(label: "org.wordpress.diskCache")
```

Comment:
I used a serial queue to reduce the number of parallel reads and writes and to ensure that the following holds true in you are using the async version of the API:

```swift
await cache.setData(data, forKey: "1")
// Cache guarantees that the data exists on disk at this point
let data = let await cache.getData(forKey: "1")
```

Global queues are concurrent.

Reply:
In https://github.com/wordpress-mobile/WordPress-iOS/pull/21059#discussion_r1261746366, I mentioned my expectation regarding `await` calls was incorrect. I think this code may have similar incorrect expectation. The guarantee in the comment can not be true. Anything can happen between these two await calls: the "1" cache may be changed to another value, it may gets deleted. Neither a serial queue nor a concurrent queue can help with that guarantee.

Given that the non-`async` versions of the getters and setters aren't guarded by a serial queue—they can be called concurrently, I don't think there is any value in using a serial queue for the `async` versions.

Reply:
> they can be called concurrently

The cache can be used by multiple subsystems with their own keys. I think it's a reasonable guarantee to provide that if you write and then read immediately after that, the value will be there.